projects/control-service/projects/helm_charts/pipelines-control-service/templates/_helpers.tpl

```diff
@@ -193,9 +197,33 @@ JDBC secret name
 {{ include "common.names.fullname" . }}-jdbc
 {{- end -}}
 
+{{/*
+VDK distribution docker repository secret name
+*/}}
+{{- define "pipelines-control-service.vdkSdkDockerRepoSecretName" -}}
+    {{ include "pipelines-control-service.fullname" . }}-vdk-sdk-docker-repo-creds
+{{- end -}}
+
+{{- define "shouldCreateVdkSdkDockerRepoSecret" }}
+  {{- if and (eq .Values.deploymentVdkDistributionImage.registryType "generic") .Values.deploymentVdkDistributionImage.registryUsernameReadOnly .Values.deploymentVdkDistributionImage.registryPasswordReadOnly }}
+    true
+  {{- end }}
+{{- end }}
+
 {{/*
 Image Pull Secret in json format
 */}}
+{{- define "buildImagePullSecretJson" }}
+    {{- $dockerRepository := index . 0 }}
+    {{- $dockerRegistryUsername := index . 1 }}
+    {{- $dockerRegistryPassword := index . 2 }}
+    {{- printf "{\"auths\": {\"%s\": {\"auth\": \"%s\"}}}" $dockerRepository (printf "%s:%s" $dockerRegistryUsername $dockerRegistryPassword | b64enc) | b64enc }}
+{{- end }}
+
 {{- define "imagePullSecretJson" }}
-{{- printf "{\"auths\": {\"%s\": {\"auth\": \"%s\"}}}" .Values.deploymentDockerRepository (printf "%s:%s" .Values.deploymentDockerRegistryUsernameReadOnly .Values.deploymentDockerRegistryPasswordReadOnly | b64enc) | b64enc }}
+    {{ include "buildImagePullSecretJson" (list .Values.deploymentDockerRepository .Values.deploymentDockerRegistryUsernameReadOnly .Values.deploymentDockerRegistryPasswordReadOnly) }}
```

Comment:
@mivanov1988  
I am very confused here. 
You seem to be passing in different values for the first parameter of the function. 
For the first secret(line 224) I see that you are passing in just the registry name. 
And for the second(line 227) you are passing in the registry name followed by the repository name. 


For me only the first solution works with the jfrog docker repo. 
Which repo requires it to have the second format?